jax/_src/random.py

```diff
@@ -2070,8 +2070,8 @@ def orthogonal(
   n = core.concrete_or_error(index, n, "The error occurred in jax.random.orthogonal()")
   z = normal(key, (*shape, n, n), dtype)
   q, r = jnp.linalg.qr(z)
-  d = jnp.diagonal(r, 0, -2, -1)
-  return lax.mul(q, lax.expand_dims(lax.div(d, abs(d).astype(d.dtype)), [-2]))
+  d = jnp.diagonal(r, axis1=-2, axis2=-1)
```

Comment:
While we're here, we may as well simplify this even further by using `d = jnp.linalg.diagonal(r)`. The `linalg` version of this function always takes the diagonal along the last two axes.